### PR TITLE
RSDK-1820 Add Validate method to module

### PR DESCRIPTION
**Pull request by benjirewis**
RSDK-1820

Adds a `Validate` method to the module service that allows modules to validate robot configs separately and return a list of implicit dependencies.

### CODE REVIEW

File: component/audioinput/v1/audioinput.pb.gw.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ import (
 
 	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
 	"github.com/grpc-ecosystem/grpc-gateway/v2/utilities"
-	"go.viam.com/api/common/v1"
+	v1_0 "go.viam.com/api/common/v1"
```
Comment (benjirewis) [inline component/audioinput/v1/audioinput.pb.gw.go:18]: These component go file changes are unrelated. Seems like the proto-compiler changed how it imports `go.viam.com/api/common/v1`? 🤷🏻

File: proto/viam/module/v1/module.proto [hunk 2]
```diff
@@ -54,3 +58,11 @@ message ReadyResponse {
   bool ready = 1;
   HandlerMap handlermap = 2;
 }
+
+message ValidateRequest {
+  viam.app.v1.RobotConfig config = 1;
```
Comment (cheukt) [inline proto/viam/module/v1/module.proto:63]: why RobotConfig?

File: component/inputcontroller/v1/input_controller.pb.gw.go [hunk 3]
```diff
@@ -15,7 +15,7 @@ import (
 
 	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
 	"github.com/grpc-ecosystem/grpc-gateway/v2/utilities"
-	"go.viam.com/api/common/v1"
+	v1_0 "go.viam.com/api/common/v1"
```
Comment (benjirewis) [inline component/inputcontroller/v1/input_controller.pb.gw.go:18]: These import name changes are unrelated 🤷🏻

File: proto/viam/module/v1/module.proto (hunk 2 above)
Comment (benjirewis) [reply proto/viam/module/v1/module.proto]: We should use `ComponentConfig`. Updated.

File: proto/viam/module/v1/module.proto [hunk 4]
```diff
@@ -20,6 +20,10 @@ service ModuleService {
 
   // Ready determines if the server is started and ready to recieve resource configurations.
   rpc Ready(ReadyRequest) returns (ReadyResponse);
+
+  // Validate determines whether the given config is valid and registers/returns implicit
+  // dependenices.
+  rpc Validate(ValidateRequest) returns (ValidateResponse);
```
Comment (Otterverse) [inline proto/viam/module/v1/module.proto:26]: Can we name this (and related functions) "ValidateConfig" for clarity?

**Review by cheukt — APPROVED** [review]
